### ISSUE TITLE
[clang] -msse2, -mavx, (etc) do not fail when not supported by the target

### ISSUE THREAD

**Issue by nekopsykose** [llvm/llvm-project#64632]
on aarch64-linux-musl:  
clang:  

```console
/ # clang main.c -msse2
clang-16: warning: argument unused during compilation: '-msse2' [-Wunused-command-line-argument]
```

gcc:
```console
/ # gcc main.c -msse2
gcc: error: unrecognized command-line option '-msse2'
```

i don't know which one is actually meant to be correct. but for configure tests that don't use -Werror ([like here](https://github.com/kdave/btrfs-progs/blob/bb0ca35245858f17a6ced97aade1dd8d70f9c9d7/configure.ac#L52)) this makes the build system think this is actually supported, and later sets a define that isn't actually then valid.

that's perhaps weird use (i never found other breakage aside from this case, so i assume usually one tests such things slightly differently or at least paired with Werror), but this specific -m value use doesn't seem like it has value being ignored instead of hard-failing?

the challenge here is disambiguating "-msse not used for some reason on x86_64 where it exists" and "-msse not used because platform doesn't have it whatsoever (different -m args for platform simd entirely)"

**Comment by MaskRay** [comment]
The errors are actually implemented on 2023-08-03 by 5fd5f8027fa196913138a34082a933abd5be0b4e :)

```
% clang --target=aarch64-unknown-linux-musl -c -mavx a.c
clang: error: unsupported option '-mavx' for target 'aarch64-unknown-linux-musl'
```

**Comment by nekopsykose** [comment]
oh nice, thanks :)
